### PR TITLE
ref(js): Minor AsyncComponent refactoring

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: static/app/components/asyncComponent.tsx [hunk 1]
```diff
@@ -444,13 +437,12 @@ export default class AsyncComponent<
...
-  shouldRenderLoading() {
-    const {loading, reloading} = this.state;
-    return loading && (!this.shouldReload || !reloading);
+  get shouldRenderLoading() {
+    return this.state.loading && (!this.shouldReload || !this.state.reloading);
   }
 
   renderComponent() {
-    return this.shouldRenderLoading()
+    return this.shouldRenderLoading
```
Comment (matejminar) [inline static/app/components/asyncComponent.tsx:445]: Should this method be called?
```suggestion
    return this.shouldRenderLoading()
```

Comment (priscilawebdev) [reply static/app/components/asyncComponent.tsx:445]: it has the get, so it's not needed

Comment (evanpurkhiser) [reply static/app/components/asyncComponent.tsx:445]: yeah I changed it to a get 🤷

**Review by priscilawebdev — APPROVED** [review]

File: static/app/components/asyncComponent.tsx (hunk 1 above)
Comment (matejminar) [reply static/app/components/asyncComponent.tsx:445]: ah cool, didn't notice that 👍

**Review by matejminar — APPROVED** [review]

**Pull request merged**